programming-core/src/main/java/org/objectweb/proactive/core/ssh/SshTunnelPool.java

```diff
@@ -200,21 +200,24 @@ public Socket getSocket(String host, int port) throws IOException {
         if (socket == null) {
             // SSH tunnel must be used
             synchronized (this.cache) {
-                int sshPort = this.config.getPort(host);
-                String username = this.config.getUsername(host);
-
-                Pair pair = this.cache.get(host);
+                String remoteHost = this.config.isHostNameDefined(sshHost) ? this.config.getHostName(sshHost) : sshHost;
```

Comment:
The check is useless here if you follow the comments above then calling `getHostName` will be enough.